source/extensions/filters/network/zookeeper_proxy/decoder.h

```diff
@@ -148,7 +148,7 @@ class DecoderImpl : public Decoder, Logger::Loggable<Logger::Id::filter> {
                                         Buffer::OwnedImpl& zk_filter_buffer);
   void decodeAndBufferHelper(Buffer::Instance& data, DecodeType dtype,
                              Buffer::OwnedImpl& zk_filter_buffer);
-  void decode(Buffer::Instance& data, DecodeType dtype);
```

Comment:
drop `&` 
we prefer to copy the integer.
do you remember why we use `uint64_t` instead of `uint32_t`?

Reply:
Will do.
> do you remember why we use `uint64_t` instead of `uint32_t`?

Not sure. But I think `uint32_t` should be enough for packet length. I could create another diff to fix it.